### DISCUSSION TITLE
Is there a way to apply a custom class to the dialog that opens up when clicking p-columnFilter?

### DISCUSSION THREAD

**Discussion by PannoSlo** [primefaces/community#2859] [PrimeNG] [answered]
Hi!
My app has two themes in it, the generic one is shown in all of the app, and the "custom" one is only applied to a restricted part of it.

I handle this by applying a custom css class to the custom section, and it generally works.
Only when i try  to apply the filter, the filter has the generic colors and styiling, because the dialog is nested under body, and not inside the div with the custom class, and rightfully so!

The fix is really simple: if i could intercept the dialog being opened, i could apply the class to the div with the "dialog" role, like we do with styleClass, when opening any dialog.

I guess my question is: can i pass something to <p-columnFilter> so that the filter dialog has a custom css class?

**Comment by qwadrox** [comment, marked as answer]
@PannoSlo Hi!

You could use javascript also to do that but I used a different approach. 

Place an empty header template where you want to style: 

```ts
<p-columnFilter
      type="text"
      field="name"
      placeholder="Search by name"
      ariaLabel="Filter Name"
    >
      <ng-template pTemplate="header"> <div class="restricted-part"></div> </ng-template>
 </p-columnFilter>
```

Target that specific filter overlay with css

```css
 ::ng-deep .p-column-filter-overlay:has(div.restricted-part) {
            background: red;
  }
```

https://stackblitz.com/edit/fuwwtk?file=src%2Fapp%2Ftable-filter-basic-demo.ts

Open the example and notice that the Country filter's style did not change - only the Name.

<img width="498" alt="image" src="https://github.com/user-attachments/assets/bc232441-3e59-4e52-be0a-dfc5fcb19776">

**Reply by PannoSlo** [reply]
Holy smokes dude!

Thank you so much, this is an awesome workaround!
